### PR TITLE
Add a sample app for the Tests API usage

### PR DESCRIPTION
**Pull request by pshevche**
## Summary

This PR adds a sample app to illustrate how the Tests API can be used to address one of the common use cases: creating a report/GitHub issues for recently unstable test classes.

### Review notes
1. A big chunk of changes is transforming the repo into a multi-module project to support the creation of multiple isolated apps.
2. Start by reviewing [`README.md`](https://github.com/gradle/gradle-enterprise-api-samples/pull/88/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5) that describes the sample app at a high level and provides an overview of the app parameters.
3. [`TestsApiSampleMain`](https://github.com/gradle/gradle-enterprise-api-samples/pull/88/files#diff-f4193d45e2ef48fb56e2f139f8c61b1ee8c7a45249e575ce091a73ea4f469573) is the entry point for the app. It contains the high-level steps to fetch the tests data of interest and delegates the reporting of unstable test containers to an instance of [`UnstableTestContainersReporter`](https://github.com/gradle/gradle-enterprise-api-samples/pull/88/files#diff-ce02ad775dc2eb949a458f665e95f5018b3e41c0c94d71b0dfa1fbde3b929f86)
4. [`StandardOutputReporter`](https://github.com/gradle/gradle-enterprise-api-samples/pull/88/files#diff-86f204af3a7daf673cb24b63136c8b2a6f3c014517a76959cf7984ec2adb7fa8) will print a small summary to the console.
5. [`GitHubCliReporter`](https://github.com/gradle/gradle-enterprise-api-samples/pull/88/files#diff-976ee4e1356c2a37a8e282c639c3fd932f91d9cba79e0304c99ffe923eb62d6a) will create a GitHub issue per unstable container in the repository of user's choice. The issue will look something like [this](https://github.com/pshevche/gradle-enterprise-api-samples/issues/5).

### CODE REVIEW

File: gradle/plugins/api-sample-app/src/main/kotlin/api-sample-app.gradle.kts [hunk 1]
```diff
@@ -84,8 +79,11 @@ openApiGenerate {
 tasks.test {
     useJUnitPlatform()
 
-    apiSpecificationURL.orNull.let { systemProperties["ge.api.url"] = it }
-
+    apiSpecificationFileGradleProperty
```
Comment (pshevche) [inline gradle/plugins/api-sample-app/src/main/kotlin/api-sample-app.gradle.kts:82]: 💬 To support executing tests against locally built API spec

File: tests-api/src/main/java/com/gradle/develocity/api/tests/BuildsQueryUtils.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.gradle.develocity.api.tests;
+
+import java.time.OffsetDateTime;
+
+final class BuildsQueryUtils {
+    private BuildsQueryUtils() {
+    }
+
+    static String projectNameEquals(String projectName) {
```
Comment (pshevche) [inline tests-api/src/main/java/com/gradle/develocity/api/tests/BuildsQueryUtils.java:9]: 💬 We can probably include more parameters, but it seemed like an overkill for this app. I just wanted to illustrate how one can combine multiple build criteria with the Tests API.

**Review by marcphilipp — COMMENTED** [review]
Instead of two separate projects, did you consider introducing different subcommands like `builds` and `tests`? I think we should compare pros and cons.

File: README.md [hunk 3]
```diff
@@ -2,7 +2,8 @@
 
 This repository demonstrates using the Gradle Enterprise API and generating client code from its OpenAPI specification.
 
-The sample observes builds being published to the given Gradle Enterprise instance in real-time and prints basic attributes along with build cache performance metrics.
+* The `builds-api` sample observes builds being published to the given Gradle Enterprise instance in real-time and prints basic attributes along with build cache performance metrics.
+* The `tests-api` sample determines previously stable test classes that have recently become unstable, and creates a report pointing to example builds published to the given Gradle Enterprise instance.
```
[the remark this answers is not in the document]
Comment (pshevche) [reply README.md:6]: Yes, we should. I just thought that there is no point in doing this now to not have a mish-mash of different names. Iurii is now working on renaming GE to Develocity in this repo. Once he is done, I will rebase and do the final pass.

[the remark this answers is not in the document]
Comment (snoopcheri) [reply README.md:6]: Makes sense. 👍

**Comment by marcphilipp** [comment]
I'm fine with either. Let's see what the @gradle/ge-build-insights team thinks who have been maintaining this repo so far.

**Comment by pshevche** [comment]
Closing it as the final version will diverge significantly, and it is easier to re-create the branch than to resolve conflicts caused by the rename

**Pull request closed without merging**